developer-docs-site/docs/tutorials/full-node/run-a-fullnode.md

```diff
@@ -0,0 +1,310 @@
+
+Given that devnet is currently being reset on a weekly basis, we estimate that Aptos will not require more than several GBs of storage. See the `#devnet-release` channel on Aptos Discord.
+
+:::
+
+## Configuring a FullNode
+
+You can configure a public FullNode in two ways:
+
+1. Using the Aptos-core source code.
+2. Using Docker.
+
+This document describes how to configure your public FullNode using both the methods.
+
+### Using Aptos-core source code
+
+1. Fork and clone the Aptos repo.
+
+    - Fork the Aptos Core repo by clicking on the **Fork** on the top right of this repo page: https://github.com/aptos-labs/aptos-core.
+    - Clone your fork.
+
+      ```
+      git clone https://github.com/<YOUR-GITHUB-USERID>/aptos-core
+
+      ```
+
+2. `cd` into `aptos-core` directory.
+
+    ```
+    cd aptos-core
+    ```
+
+3. Run the `scripts/dev_setup.sh` Bash script as shown below. This will prepare your developer environment.
+
+    ```
+    ./scripts/dev_setup.sh
+    ```
+
+4. Update your current shell environment.
+
+    ```
+    source ~/.cargo/env
+    ```
+
+With your development environment ready, now you can start to setup your FullNode.
+
+5. Checkout the `devnet` branch using `git checkout --track origin/devnet`.
+
+  In the next step you will download and edit the configuration YAML file `public_full_node.yaml` to ensure that your FullNode:
+
+    - Contains the correct genesis blob that is published by the Aptos devnet.
+    - Synchronizes correctly with the devnet, by using the checkpoint file `waypoint.txt` published by the devnet, and
+    - Stores the devnet database at a location of your choice on your local machine.
+
+6. Make sure your current working directory is `aptos-core`. Copy the YAML configuration file from `config/src/config/test_data/public_full_node.yaml` to your current working directory. You will edit this file.
+
+7. The Aptos devnet publishes the `genesis.blob` and `waypoint.txt` files. Download them:
+
+    - Click here [genesis][devnet_genesis] or run the below command on your terminal:
+      ```
+      wget https://devnet.aptoslabs.com/genesis.blob
+      ```
+
+    - Click here [waypoint][devnet_waypoint] and save the file, or run the below command on your terminal:
+      ```
+      wget https://devnet.aptoslabs.com/waypoint.txt
+      ```
+
+8. Edit the `public_full_node.yaml` file in your current working directory as follows. See the example YAML file in: `docker/compose/public_full_node/public_full_node.yaml`.
+
+    - For the `data_dir` key in the `base` list, specify the directory where on your local computer you want to store the devnet database. This can be anywhere on your computer. For example, you can create a directory `my-full-node/data` in your home directory and specify it as:
+
+      ```
+      data_dir: "/path/to/my/homedir/my-full-node/data"
+      ```
+
+    - Copy and paste the contents of the `waypoint.txt` file into the `from_config` field of the `waypoint` list. For example:
```

Comment:
Users might think based on the previous step that this should go in the `data` folder, but you actually just want it in the root (as in, in `my-full-node`).

Perhaps re-order these steps. Since the previous thing just talked about genesis and waypoint, do this and the next one and then add the data_dir step as the third one.